plasmapy/diagnostics/proton_radiography.py

```diff
@@ -300,6 +305,212 @@ def _log(self, msg):
+            extent = (extent,)
+
+        if len(extent) == 1:
+            radius = 0.5 * extent[0].si.value
+            width = extent[0].si.value
+            height = extent[0].si.value
+        elif len(extent) == 2:
+            radius = None
+            width = extent[0].si.value
+            height = extent[1].si.value
+        else:
+            raise ValueError(
+                "extent must be a tuple of 1 or 2 elements, but "
+                f"{len(extent)} elements were provided."
+            )
+
+        if not isinstance(nwires, tuple):
+            nwires = (nwires,)
+
+        if len(nwires) != 2:
+            nwires = (nwires[0], nwires[0])
+
+        # If no hdir/vdir is specified, calculate a default value
+        # If one is specified, make sure it is normalized
+        if mesh_hdir is None:
+            # Re-calculate the default here, in case the user
+            # specified a different det_hdir
+            mesh_hdir = self._default_detector_hdir()
+        else:
+            mesh_hdir = mesh_hdir / np.linalg.norm(mesh_hdir)
+
+        if mesh_vdir is None:
+            mesh_vdir = np.cross(mesh_hdir, self.det_n)
+            mesh_vdir = -mesh_vdir / np.linalg.norm(mesh_vdir)
+        else:
+            mesh_vdir = mesh_vdir / np.linalg.norm(mesh_vdir)
+
+        # Raise exception if mesh is AFTER the field grid
+        if np.linalg.norm(location - self.source) > np.linalg.norm(self.source):
+            raise ValueError(
+                f"The specified mesh location, {location},"
+                "is not between the source and the origin."
+            )
+
+        mesh_entry = {
+            "location": location,
+            "wire_radius": wire_radius,
+            "radius": radius,
+            "width": width,
+            "height": height,
+            "nwires": nwires,
+            "mesh_hdir": mesh_hdir,
+            "mesh_vdir": mesh_vdir,
+        }
+
+        self.mesh_list.append(mesh_entry)
+
+    def _apply_wire_mesh(
+        self,
+        location=None,
+        wire_radius=None,
+        radius=None,
+        width=None,
+        height=None,
+        nwires=None,
+        mesh_hdir=None,
+        mesh_vdir=None,
+    ):
+        """
+        Apply wire meshes that were added to self.mesh_list
+        """
+        x = self._coast_to_plane(location, mesh_hdir, mesh_vdir)
+
+        # Particle positions in 2D on the mesh plane
+        xloc = np.dot(x - location, mesh_hdir)
+        yloc = np.dot(x - location, mesh_vdir)
+
+        # Create an array in which 0 indicates that a particle has hit one of
+        # the wires and any other value indicates that it has not
+        no_hit = np.ones([self.nparticles])
+
+        # Mark particles that overlap vertical or horizontal position with a wire
+        h_centers = np.linspace(-width / 2, width / 2, num=nwires[0])
+        for c in h_centers:
+            no_hit *= np.where(np.isclose(xloc, c, atol=wire_radius), 0, 1)
```

Comment:
For booleans this will probably simplify to something like... (Not quite this; just sketching)
```suggestion
            no_hit &= ~np.isclose(xloc, c, atol=wire_radius)
```

Probably easier to reason about this if you flip the condition, start with all falses and apply "or" with the isclose for each wire.

Reply:
NICE